transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/httpx/ExtendedHandshakeIT.java

```diff
@@ -0,0 +1,150 @@
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/path"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .acceptOption("ws.inactivity.timeout", "123secs")
+                        .done()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/auth"))
+                            .type("echo")
+                            .realmName("Kaazing WebSocket Gateway Demo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .authorization()
+                                .requireRole("AUTHORIZED")
+                            .done()
+                            .acceptOption("ws.inactivity.timeout", "123secs")
+                        .done()
+                        .security()
+                            .realm()
+                                .name("Kaazing WebSocket Gateway Demo")
+                                .description("Kaazing WebSocket Gateway Demo")
+                                .httpChallengeScheme("Application Basic")
+                                .authorizationMode("challenge")
+                                .loginModule()
+                                    .type("class:org.kaazing.gateway.security.auth.YesLoginModule")
+                                    .success("requisite")
+                                    .option("roles", "AUTHORIZED")
+                                .done()
+                            .done()
+                        .done()
+                    .done();
+            init(configuration);
+        }
+    };
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, k3po);
+
+    @Test
+    @Specification({ "connection.established.with.authorization/request" })
+    public void shouldEstablishConnectionWithAuthorization() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({ "connection.established/request" })
+    public void shouldEstablishConnection() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({"connection.established.data.exchanged.close/request"})
+    public void shouldEstablishConnectionAndExchangeDataAndClose() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({ "client.sends.message.between.opening.and.extended.handshake/request" })
+    public void shouldFailWhenClientSendsMessageBetweenOpeningAndExtendedHandshake() throws Exception {
+        k3po.finish();
+    }
+
```

Comment:
Sorry I forgot to mention, I usually also put a comment where the @Test would usually be saying 
````
// Client only test
```